### ISSUE TITLE
LPA can't attack normal training.

### ISSUE THREAD

**Issue by kevinbro96** [cassidylaidlaw/perceptual-advex#4]

**Comment by kevinbro96** [comment]
I train a model with clean data using the following code:
python adv_train.py --batch_size 128 --arch resnet50 --dataset cifar --attack "NoAttack()" --only_attack_correct.
The LagrangePerceptualAttack accuracy is 45% which is different from 0% in line 1 of Table 3.

**Comment by kevinbro96** [comment]
The whole results:
END EPOCH 0099
BEGIN EVALUATION
ATTACK NoAttack accuracy: 95.0%
ATTACK LinfAttack accuracy: 0.0%
ATTACK L2Attack accuracy: 0.0%
ATTACK JPEGLinfAttack accuracy: 0.0%
ATTACK FogAttack accuracy: 7.6%
ATTACK StAdvAttack accuracy: 0.0%
ATTACK ReColorAdvAttack accuracy: 0.7%
ATTACK LagrangePerceptualAttack accuracy: 70.5%
END EVALUATION

**Comment by cassidylaidlaw** [comment]
The training code evaluates using self-bounded LPA, which is weak for normally trained ResNet-50s. In the paper, we evaluate with AlexNet-bounded LPA (using AlexNet for the LPIPS distance). You could try running this command to evaluate your trained model with both self-bounded and AlexNet-bounded LPA:
```
python evaluate_trained_model.py --dataset cifar --checkpoint /path/to/checkpoint.pt --arch resnet50 --batch_size 100 --output evaluation.csv \
"NoAttack()" \
"LagrangePerceptualAttack(model, num_iterations=40, bound=0.5, lpips_model='self', projection='newtons')" \
"LagrangePerceptualAttack(model, num_iterations=40, bound=0.5, lpips_model='alexnet_cifar', projection='newtons')"
```
Running this on a normally trained ResNet-50, I get ~95% clean accuracy, ~60% accuracy against self-bounded LPA, and 0% accuracy against AlexNet-bounded LPA.

**Comment by kevinbro96** [comment]
Thanks for the reply.
